contributor-guide/modules/ROOT/pages/design-guide/dependencies.adoc

```diff
@@ -3,74 +3,46 @@
- header includes only the lightweight headers
- <boost/config.hpp> and <cstddef>. There are no
- runtime costs at all. With costs so low and benefits so high,
- other boost libraries should use boost::noncopyable when the
- need arises except in exceptional circumstances.
+Other libraries you might consider as good foundational components include:
 
+* boost:utility[] includes several non-templated, non-data structure related classes and functions, such as base-from-member idiom, checked delete, next and prior functions, noncopyable, and result_of.
 
-**Example where a standard library component might
- possibly be used:** Providing diagnostic output as a
- debugging aid can be a nice feature for a library. Yet using
- Standard Library <iostream> can involve a lot of
- additional cost, particularly if <iostream> is unlikely
- to be used elsewhere in the application. In certain GUI or
- embedded applications, coupling to <iostream> would be a
- disqualification. Consider redesign of the boost library in
- question so that the user supplies the diagnostic output
- mechanism.
+* boost:concept-check[] provides tools for specifying and checking that types meet the requirements of generic algorithms. It's used by many other libraries to ensure that template parameters meet the necessary requirements.
 
+* boost:type-traits[] provides tools for introspecting the properties of types, and is used extensively throughout Boost. It offers a collection of templates for querying and manipulating type information at compile-time.
 
-**Example where another boost component should not be
- used:** The boost dir\_it library has considerable
- coupling and runtime costs, not to mention portability issues
- for unsupported operating systems. While completely appropriate
- when directory iteration is required, it would not be
- reasonable for another boost library to use dir\_it just to
- check that a file is available before opening. C++ Standard
- Library file open functionality does this at lower cost. Don't
- use dir\_it just for the sake of using a boost library.
+* For handling data types and structures, one of boost:function-types[], boost:fusion[], boost:any[], boost:variant[] might provide what you need.
 
+* For metaprogramming, boost:mpl[] or boost:preprocessor[] provide many of the building blocks for this style of programming.
 
+== Examples of Questionable Dependencies
+
+An example where a standard library component might be used, but with care, is in providing diagnostic output. This output is used as a debugging aid and can be a nice feature for a library. Yet using Standard Library `<iostream>` can involve a lot of additional cost, particularly if `<iostream>` is unlikely to be used elsewhere in the library users' application. In certain GUI or embedded applications, coupling to `<iostream>` would be a disqualification. In this case, consider a redesign of the boost library in question so that the user supplies the diagnostic output mechanism.
```

Comment:
> An example where a standard library component might be used, but with care

We don't have particular restrictions against "standard library components" at all in boost. We can't make it sound like being allowed to use the standard library is an exception.

Some standard headers should be avoided for multiple reasons in C++ but these reasons are not particular to Boost.

Reply:
OK, removed the reference to <iostream>